--- etc/client.cfg.dist.orig	2011-03-08 07:27:20.123730986 -0500
+++ etc/client.cfg.dist	2011-03-08 07:28:00.823106115 -0500
@@ -1,17 +1,38 @@
-topicprefix = /topic/mcollective
-libdir = /usr/libexec/mcollective
-logfile = /dev/null
+# It's recommended that you do not set host, user, password, or psk
+# in this file and instead use environment variables, such as the
+# following:
+#
+#   export STOMP_USER=user
+#   export STOMP_PASSWORD=password
+#   export STOMP_SERVER=stomp.your.com
+#   export MCOLLECTIVE_PSK=123456789
+#
+# If you go this route, you may leave the variables below with "unset" as
+# the value.
+# Topic to subscribe to
+topicprefix = /topic/
+main_collective = mcollective
+collectives = mcollective
+
+# Plugin location
+libdir = /usr/share/mcollective
+
+# Client logfile
+logfile = /var/log/mcollective.log
+
+# Client loglevel
 loglevel = info
 
 # Plugins
 securityprovider = psk
 plugin.psk = unset
 
+# Connector Information
 connector = stomp
-plugin.stomp.host = stomp.example.com
+plugin.stomp.host = unset
 plugin.stomp.port = 6163
-plugin.stomp.user = mcollective
-plugin.stomp.password = marionette
+plugin.stomp.user = unset
+plugin.stomp.password = unset
 
 # Facts
 factsource = yaml
--- etc/server.cfg.dist.orig	2011-03-08 07:27:24.703371306 -0500
+++ etc/server.cfg.dist	2011-03-08 07:28:12.973133167 -0500
@@ -1,18 +1,31 @@
-topicprefix = /topic/mcollective
-libdir = /usr/libexec/mcollective
-logfile = /var/log/mcollective.log
+#
+# THIS FILE SHOULD NOT BE WORLD READABLE
+#
+topicprefix = /topic/
+main_collective = mcollective
+collectives = mcollective
+
+# Plugins location
+libdir = /usr/share/mcollective
+
+# Server logfile
+logfile = /var/log/mcollectived.log
+
+# Server loglevel
 loglevel = info
+
+# Set this to 1 to background the process
 daemonize = 1
 
 # Plugins
 securityprovider = psk
-plugin.psk = unset
+plugin.psk = SET_A_PSK_HERE
 
 connector = stomp
-plugin.stomp.host = stomp.example.com
+plugin.stomp.host = SET_STOMP_SERVER
 plugin.stomp.port = 6163
-plugin.stomp.user = mcollective
-plugin.stomp.password = marionette
+plugin.stomp.user = SET_STOMP_USERNAME
+plugin.stomp.password = SET_STOMP_PASSWORD
 
 # Facts
 factsource = yaml
