--- lib/puppet/util/monkey_patches.rb.orig	2010-06-04 03:03:34.000000000 +0200
+++ lib/puppet/util/monkey_patches.rb	2010-06-04 03:04:05.000000000 +0200
@@ -7,2 +7,15 @@
     end
 end
+
+#
+# Workaround for bug in MRI 1.8.7, see
+#     http://redmine.ruby-lang.org/issues/show/2708
+# for details
+#
+if RUBY_VERSION == '1.8.7'
+    class NilClass
+        def closed?
+            true
+        end
+    end
+end
